### PR TITLE
Features/#366 adjust time dependent tables

### LINKED ISSUE #366: lines_t and links_t must be adjusted in the eHV clustering
The eHV clustering is not considering the temporal series in lines_t and links_t. This problem is not in the kmean clustering. Checking the code which deals with these tasks in the Pypsa libraries is recommended.

### PR DESCRIPTION
**Pull request by CarlosEpia**
Fixes #366 and #374

### CODE REVIEW

**Review by ulfmueller — CHANGES_REQUESTED** [review]
This implementation does not work for the [current pypsa version we are using](https://github.com/openego/eTraGo/blob/features/%23366-adjust-time-dependent-tables/setup.py#L34) 
I would wait for #369 to be implemented and pull that then in here to ensure that it all works fine. 

Other than that, it looks very good and useful!

**Comment by ClaraBuettner** [comment]
> This implementation does not work for the [current pypsa version we are using](https://github.com/openego/eTraGo/blob/features/%23366-adjust-time-dependent-tables/setup.py#L34) I would wait for #369 to be implemented and pull that then in here to ensure that it all works fine.

The pypsa update was merged into `dev`.

File: etrago/cluster/networkclustering.py [hunk 1]
```diff
@@ -267,7 +269,7 @@ def strategies_links():
         "p_nom": np.sum,
         "p_nom_extendable": _make_consense_links,
         "p_nom_max": np.sum,
-        "capital_cost": np.mean,
+        "capital_cost": np.sum,
```
[the remark this answers is not in the document]
Comment (CarlosEpia) [reply etrago/cluster/networkclustering.py]: Yeah. It makes more sense to keep the aggregation function for the capital_cost as the mean. it was already changed in the  PR d655fb8de85a5d2ef54eba062a67f00d88d67bba

**Comment by ClaraBuettner** [comment]
@CarlosEpia:
Could you change the order of these lines: 
https://github.com/openego/eTraGo/blob/9ee676556e2e71213b576de02db83968e599141f/etrago/cluster/gasclustering.py#L217-L218

**Review by ulfmueller — CHANGES_REQUESTED** [review]
Unfortunately I 'have to' ask some questions:
1. For me the k-means clustering is not working in this branch. I get a consense error in the aggregatelines function: 
```
File "/home/ulf/venvs/etrago/lib/python3.7/site-packages/pypsa/networkclustering.py", line 203, in aggregatelines
    lines = interlines_c.groupby(['bus0_s', 'bus1_s']).apply(aggregatelinegroup)

  File "/home/ulf/venvs/etrago/lib/python3.7/site-packages/pandas/core/groupby/groupby.py", line 1275, in apply
    result = self._python_apply_general(f, self._selected_obj)

  File "/home/ulf/venvs/etrago/lib/python3.7/site-packages/pandas/core/groupby/groupby.py", line 1309, in _python_apply_general
    keys, values, mutated = self.grouper.apply(f, data, self.axis)

  File "/home/ulf/venvs/etrago/lib/python3.7/site-packages/pandas/core/groupby/ops.py", line 841, in apply
    res = f(group)

  File "/home/ulf/venvs/etrago/lib/python3.7/site-packages/pypsa/networkclustering.py", line 200, in aggregatelinegroup
    data.update((f, consense[f](l[f])) for f in columns.difference(data))

  File "/home/ulf/venvs/etrago/lib/python3.7/site-packages/pypsa/networkclustering.py", line 200, in <genexpr>
    data.update((f, consense[f](l[f])) for f in columns.difference(data))

  File "/home/ulf/venvs/etrago/lib/python3.7/site-packages/pypsa/networkclustering.py", line 51, in consense
    .format(component, x.name, attr, x)

AssertionError: In Bus cluster carrier the values of attribute carrier do not agree:
Line
1107      
1108      
1109
```
2. I could not verify #374 therefore (but code-wise it looks fine)
3. in the ehv clustering I checked the lines and lines_t.s_max_pu and I was wondering why apparently not all lines have these _t.s_max_pu values. Why is that? 
4. The different data frames in the links_t object are not callable. Apparently the links_t becomes a dict and really should be a pypsa dict. Anyhow this unexpected type leads to an error in the lopf. This seems to be a real issue and should be addressed before merging.

**Comment by CarlosEpia** [comment]
Hi @ulfmueller,
About your first question, can I ask which db (creation date) and pypsa version are you using ?
Regarding the number of lines and time series, there is a problem in the input that must be solved. https://github.com/openego/eGon-data/issues/645.
About the dict and pypsa dict, I will take care of it as soon as possible.

**Comment by ulfmueller** [comment]
Data from the 6th of January, pypsa 0.19.1

**Comment by CarlosEpia** [comment]
@ulfmueller would you mind to use the db called "sh-egon-data-grid-0702.backup" available at: https://next.rl-institut.de/s/JrpDkGDDbRQoBBM?
That data was generated about 3 weeks ago and I just performed a successful test using it. Hopefully, it would work for you too.

**Comment by ulfmueller** [comment]
That works for the pure ehv clustering! Great! 

In context of checking #374 I noticed s.th. else now: After the ehv_clustering there are dsm-xyz (e.g. dsm-cts) carrier buses which have a v_nom of 110 kV. First of all, the dsm-buses should be non-AC, so should not have any v_nom, right? Second, it definitely looks weird if there are still 110 kV buses after the ehv clustering!

Actually trying to check #374 I still run into a consense error which links back to:

`File "/home/ulf/github/eTraGo/etrago/cluster/networkclustering.py", line 324, in group_links
    new_df = links.groupby(grouper, axis=0).agg(strategies)`

and gets called here with the following AssertionError:

`File "/home/ulf/github/eTraGo/etrago/cluster/networkclustering.py", line 216, in _make_consense_links
    ).all() or x.isnull().all(), f"No consense in table links column {x.name}: \n {x}"
AssertionError: No consense in table links column country: 
 115    GBBE
119    GBNL
Name: country, dtype: object`

**Comment by ulfmueller** [comment]
- For the newest data set the lopf is working after the k-means clustering as well. Great!
- The dsm still has 110 kV after the k-means clustering. It looks a bit weird but maybe we can leave like that since they are modelled as non-AC components but that is where they originally are connected to the AC system. Maybe we can discuss that in our regular meeting. But for now, I guess we can leave like that.
- The non-AC buses have a v_nom of 1 kV which is the pypsa default. That is a bit weird but without effect, I would say. I would be fine to leave it like that right now. In p-e-s the same thing happens so far for our p-e-s results.
- Now when using only the ehv clustering an error occurs, 'at the very end' after the lopf got successfully solved:

```
  File "/home/ulf/venvs/etrago/lib/python3.7/site-packages/pypsa/opf.py", line 1316, in extract_optimisation_results
    set_from_series(network.links_t.p0, get_values(model.link_p))

AttributeError: 'dict' object has no attribute 'p0'
```
@CarlosEpia Do you get that one as well?

**Comment by CarlosEpia** [comment]
Hi @ulfmueller 
Regarding the non AC buses with assigned voltage, I agree with you. We can discuss it later since it is not an urgent topic.
I reproduced the error you described when using the eHV clustering alone. It was solved in ca78a8cba2cdc60776578c742c4d97bbc510525a

**Review by ulfmueller — APPROVED** [review]

**Pull request merged**